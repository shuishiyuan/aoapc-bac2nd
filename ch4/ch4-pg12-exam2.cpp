#include <stdio.h>

int main() {
    printf("Hangman Judge, UVa489\n\n");
    int rnd;
#ifdef LOCAL
    freopen("./data/ch4-pg12-exam2.in", "r", stdin);
#endif
    while(scanf("%d\n", &rnd) == 1 && rnd != -1) {
        int n[26] = {0};
        int cnt = 0;
        int wronger = 0;
        char c;
        while((c = getchar()) != '\n') {
            if (n[c - 'a']++ == 0) {
                cnt++;
            }
        }
        while((c = getchar()) != '\n') {
            if (cnt > 0 && wronger < 7) {
                if (n[c - 'a'] > 0) {
                    cnt--;
                    n[c - 'a'] = 0;
                } else {
                    wronger++;
                }
            }
        }
        printf("Round %d\n", rnd);
        if (cnt == 0) {
            printf("You Win.\n");
        } else if (wronger == 7) {
            printf("You Lose.\n");
        } else {
            printf("You chickened out.\n");
        }
    }
    return 0;
}
