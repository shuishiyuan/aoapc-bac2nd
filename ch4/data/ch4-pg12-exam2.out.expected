Round 1
You Win.
Round 2
You chickened out.
Round 3
You Lose.
Round 4
You Lose.

